lib/tree.rb

```diff
@@ -1,64 +1,126 @@
 class TreeNode
   attr_reader :key, :value
   attr_accessor :left, :right
-
-   def initialize(key, val)
+  
+  def initialize(key, val)
     @key = key
     @value = val
     @left = nil
     @right = nil
-   end
+  end
 end
 
 class Tree
   attr_reader :root
   def initialize
     @root = nil
   end
-
-  # Time Complexity: 
-  # Space Complexity: 
+  
+  # Time Complexity: O(log n)
+  # Space Complexity: O(1)
+  def add_helper(current_node, key, value)
```

Comment:
The space complexity is O(log n) if the tree is balanced and O(n) if it's not.

Reply:
Fixed.